src/mscorlib/shared/System/Threading/Tasks/TaskCanceledException.cs

```diff
@@ -53,6 +53,18 @@ public TaskCanceledException(string message, Exception innerException) : base(me
         {
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="T:System.Threading.Tasks.TaskCanceledException"/>
+        /// class with a specified error message, a reference to the inner exception that is the cause of
+        /// this exception, and the <see cref="CancellationToken"/> that triggered the cancellation.
+        /// </summary>
+        /// <param name="message">The error message that explains the reason for the exception.</param>
+        /// <param name="innerException">The exception that is the cause of the current exception.</param>
+        /// <param name="token">The <see cref="CancellationToken"/> that triggered the cancellation.</param>
+        public TaskCanceledException(string message, Exception innerException, CancellationToken token) : base(message, innerException, token)
+        {
```

Comment:
Do you see any scenarios where it is interesting to also include the Task with these arguments? If so it might be a good idea to add that overload as well.

Reply:
It's a good question, but scenario-wise, not really.  I can't remember ever needing to construct one of these until now, and now only because I want the type identity but basically to construct it as if it's an OperationCanceledException.